### PR TITLE
Add Sync Text Reveal to the Playing Voice

### PR DESCRIPTION
**Pull request by CakeVR**
This PR adds the ability to slow down text to match the current character speaking.

Many Visual Novels have this feature, it enforces the player's reading to slowdown, which may be wanted for immersion. Especially if they want to wait for the voice to finish playing.

Closes https://github.com/dialogic-godot/dialogic/issues/1908.

### CODE REVIEW

**Comment by Pheubel** [comment]
do you have some examples of it in action?

**Comment by CakeVR** [comment]
> do you have some examples of it in action?

I have no video of this feature. However, it simply slows down the letter speed based on the current voice audio, if there is any.

File: addons/dialogic/Modules/Text/subsystem_text.gd [hunk 1]
```diff
@@ -209,33 +224,65 @@ func hide_next_indicators(_fake_arg = null) -> void:
...
 
 
-func update_text_speed(letter_speed:float = -1, absolute:bool = false, _speed_multiplier:float= -1, _user_speed:float=-1) -> void:
+## This method will sync the text speed to the voice audio clip length, if a
+## voice is playing.
+## For instance, if the voice is playing for four seconds, the text will finish
+## revealing after this time.
+## This feature ignores Auto-Pauses on letters. Pauses via BBCode will desync
+## the reveal.
+func set_voice_synced_text(enabled: bool) -> void:
```
Comment (Jowan-Spooner) [inline addons/dialogic/Modules/Text/subsystem_text.gd:233]: I suggest `set_text_voice_synced(enabled := true)`
I think it's common for methods like these to be true by default. Also your name makes it sounds as if you set a text.

File: addons/dialogic/Modules/Text/subsystem_text.gd [hunk 2]
```diff
@@ -209,33 +224,65 @@ func hide_next_indicators(_fake_arg = null) -> void:
...
+	_voice_synced_text = enabled
+	update_text_speed()
+
+
+## Returns whether voice-synced text is enabled.
+func get_voice_synced_text() -> bool:
```
Comment (Jowan-Spooner) [inline addons/dialogic/Modules/Text/subsystem_text.gd:239]: I suggest `is_text_voice_synced() -> bool`
Starting bool getters with is_ is the standard, at least in godot.

File: addons/dialogic/Modules/Text/subsystem_text.gd [hunk 3]
```diff
@@ -209,33 +224,65 @@ func hide_next_indicators(_fake_arg = null) -> void:
...
-	else:
-		speed_multiplier = _speed_multiplier
-
-	if _user_speed == -1:
-		_user_speed = dialogic.Settings.get_setting('text_speed', 1)
+	_voice_synced_text = reveal_to_playing_voice
 
 
 	for text_node in get_tree().get_nodes_in_group('dialogic_dialog_text'):
+		text_node.sync_reveal_to_voice = reveal_to_playing_voice
```
Comment (Jowan-Spooner) [inline addons/dialogic/Modules/Text/subsystem_text.gd:274]: This property has been renamed on the DialogText node

**Pull request merged**